File: V2/findrey-server/server/core/dto/transaction_dto/response.go
package transaction_dto

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

type TransactionResponseDTO struct {
	ID                 primitive.ObjectID `json:"id"`
	UserID             primitive.ObjectID `json:"userId"`
	TransactionDate    time.Time          `json:"transactionDate"`
	TransactionType    primitive.ObjectID `json:"transactionType"`
	SubTransactionType primitive.ObjectID `json:"subTransactionType,omitempty"`
	DebitAccount       primitive.ObjectID `json:"debitAccount"`
	CreditAccount      primitive.ObjectID `json:"creditAccount,omitempty"`
	Category           primitive.ObjectID `json:"category"`
	Description        string             `json:"description"`
	Amount             float64            `json:"amount"`
	CreatedAt          time.Time          `json:"createdAt"`
	UpdatedAt          time.Time          `json:"updatedAt"`
}

File: backend/app/core/schema/Transactions/Response.py
from datetime import datetime

from pydantic import BaseModel


class Response(BaseModel):
    transactionId: int
    user_id: str
    transactionDate: str
    transactionType: str
    subTransactionType: str
    debitAccount: str
    creditAccount: str
    category: str
    description: str
    amount: float
